common/SC_PaUtils.hpp

```diff
@@ -0,0 +1,167 @@
+// helper functions for SuperCollider server's PortAudio backend
+
+#ifndef SC_PAUTILS_HPP
+#define SC_PAUTILS_HPP
+
+#include "portaudio.h"
+#include <stdlib.h>
+
+enum class IOType { Input, Output };
+
+static std::string GetDeviceName(int index) {
```

Comment:
instead of being `static`, I would much rather see all function definitions here moved to a cpp file (note that you will need to add the file [in CMakeLists](https://github.com/supercollider/supercollider/blob/develop/server/scsynth/CMakeLists.txt#L52) for each target you want to use it in)

when you do that, you should audit whether all of these functions actually need to be exposed

you should also add some brief documentation for each function

Reply:
> when you do that, you should audit whether all of these functions actually need to be exposed

The purpose of this PR is to expose functions that are to be shared between PA backends in scsynth and supernova. Do you mean I should check whether they will indeed be needed for both synths?

Reply:
i mean, do all these function declarations need to be put in the header, or can some stay encapsulated in the cpp file? in other words, are some of them only called within other functions in this header? we should only expose what's absolutely necessary, and encapsulate the rest